PythonVisualizations/Array.py

```diff
@@ -258,9 +258,54 @@ def remove(self, val):
             return True
         return False
 
+    def traverse(self):
+        self.cleanUp()
+
+        # draw an index pointing to the first cell
+        indexDisplay = self.createIndex(0, 'j')
+        self.cleanup |= set(indexDisplay)
+
+        # draw output box
+        canvasDimensions = self.widgetDimensions(self.canvas)
+        outputBox = self.canvas.create_rectangle(\
+            ARRAY_X0+ CELL_SIZE//2, ARRAY_Y0 + canvasDimensions[1] // 3, ARRAY_X0 + CELL_SIZE*len(self.list), \
+            ARRAY_Y0 + canvasDimensions[1] // 3 + 50,\
+            fill = OPERATIONS_BG)
+        self.cleanup.add(outputBox)
+
+        for j in range(len(self.list)):
+            # calculate where the value will need to move to
+            outputBoxCoords = self.canvas.coords(outputBox)
+            midOutputBox = (outputBoxCoords[3]-outputBoxCoords[1])//2+outputBoxCoords[1]
+
+            # create the value to move to output box
+            valueOutput = self.copyCanvasItem(self.list[j].display_val)
+            valueList = (valueOutput,)
+            self.cleanup.add(valueOutput)
+
+            # move value to output box
+            toPositions = (outputBoxCoords[0] + (j+1)*(CELL_SIZE*3)//4, midOutputBox)

```

Comment:
With the outputBox as specified above, the placement of the text items might be better as:
```
            toPositions = (outputBoxCoords[0] + padding/2 + (j+1/2)*spacing,
                           midOutputBox)
```